### PR TITLE
feat: add JUnit reporter

### PR DESCRIPTION
**Pull request by Demivan**
Tested on GitLab:
image
image
image

### CODE REVIEW

**Comment by Demivan** [comment]
@patak-dev 
I have no idea why d34888d is needed. Did a `git bisect` and found that moving `flattenTasks` out to be reused causes a typecheck error.

File: packages/vitest/src/node/reporters/junit.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        errors: 0, // An errored test is one that had an unanticipated problem. We cannot detect those.
+        skipped: file.stats.skipped,
+        time: getDuration(file),
+      }))
+
+      this.logger.indent()
+      await this.writeTasks(file.tasks, file.name)
+      this.logger.unindent()
+
+      await this.logger.log('</testsuite>')
```
Comment (patak-dev) [inline packages/vitest/src/node/reporters/junit.ts:189]: Would it make sense to have a `logElement` function that will abstract the indent, unindent and closing of tags to simplify this reporter?
```ts
  logElement: async function(name: string, attrs: Record<string, any>, children: () => void) {
name: string, attrs: Record<string, any>) {
    await this.logger.log(openTag(name, attrs))
    this.logger.indent()
    await children.call(this)
    this.logger.unindent()

    await this.logger.log(`</${name}>`)
  }
```
So you could call `logElement` instead here (and in other places where you have `openTag`, indent, etc
```ts
      await this.logElement('testsuite', {
        name: file.name,
        timestamp: (new Date()).toISOString(),
        hostname: hostname(),
        tests: file.tasks.length,
        failures: file.stats.failures,
        errors: 0, // An errored test is one that had an unanticipated problem. We cannot detect those.
        skipped: file.stats.skipped,
        time: getDuration(file),
      }, () => {
        await this.writeTasks(file.tasks, file.name)
      })
```

Comment (Demivan) [reply packages/vitest/src/node/reporters/junit.ts:189]: Makes sense. Will implement this refactoring tomorrow.

**Review by patak-dev — APPROVED** [review]

**Comment by patak-dev** [comment]
Awesome work @Demivan, enabling Vitest for gitlab users should help a lot with adoption

**Comment by Demivan** [comment]
Thanks, this will allow us to finish migration to Vitest. JUnit reporter was the last missing feature.

**Comment by Demivan** [comment]
Waiting for release :slightly_smiling_face:

**Comment by snake-py** [comment]
Hey I am new to this package and I am also trying to make this work how can I include Junit as a reporter? I tried to added to the reporter list but this throws me an error. 

```
Serialized Error: {
  "code": "MODULE_NOT_FOUND",
  "requireStack": [
    "/home/fabio/workdigital/packages/component-library-react/node_modules/istanbul-reports/index.js",
    "/home/fabio/workdigital/packages/component-library-react/node_modules/c8/lib/report.js",
  ],
}
```

**Pull request merged**